cmd/tikv-ctl/src/main.rs

```diff
@@ -1955,10 +1956,160 @@ fn main() {
         return;
     }
 
-    // Deal with all subcommands needs PD.
-    if let Some(pd) = opt.pd.as_deref() {
-        let pd_client = get_pd_rpc_client(pd, Arc::clone(&mgr));
-        if let Cmd::CompactCluster {
+    if matches!(cmd, Cmd::CompactCluster { .. } | Cmd::SplitRegion { .. }) {
```

Comment:
Why bother to check command type and pd here rather than in the command-handling block? We can embed the `--pd` checking into the `get_pd_rpc_client` like this: `let pd_client = get_pd_rpc_client(&*opt.pd, Arc::clone(&mgr))?;`

Reply:
If so, we can avoid repeating the command type twice.

Reply:
> If so, we can avoid repeating the command type twice.

Fixed.